QEfficient/transformers/quantizers/quantizer_gptq.py

```diff
@@ -0,0 +1,152 @@
+
+        Args:
+            quantization_config (QEffGPTQConfig): The quantization configuration.
+            **kwargs: Additional keyword arguments.
+        """
+        super().__init__(quantization_config, **kwargs)
+
+    def validate_environment(self, device_map, **kwargs):
+        """
+        Validates the environment for quantization.
+
+        Args:
+            device_map (dict): The device map for the model.
+            **kwargs: Additional keyword arguments.
+
+        Returns:
+            :bool: True if the environment is valid, False otherwise.
+        """
+        return True
+
+    def update_torch_dtype(self, torch_dtype):
+        """
+        Updates the torch data type for quantization.
+
+        Args:
+            torch_dtype (torch.dtype): The requested torch data type.
+
+        Returns:
+            :torch.dtype: The updated torch data type.
+        """
+        if torch_dtype not in [None, torch.float32]:
+            logger.warning(f"Requested dtype {torch_dtype} is not supported, overriding to None")
+        return None
+
+    def _process_model_before_weight_loading(self, model, **kwargs):
+        """
+        Processes the model before loading weights, ensuring it is suitable for quantization.
+
+        Args:
+            model (torch.nn.Module): The model to process.
+            **kwargs: Additional keyword arguments.
+
+        Returns:
+            :torch.nn.Module: The processed model.
+        """
+        if model.__class__.main_input_name != "input_ids":
+            raise RuntimeError("We can only quantize pure text model.")
+        if not self.pre_quantized:
+            raise RuntimeError("Model is not quantized")
+
+        self.modules_to_not_convert = get_keys_to_not_convert(model)
+
+        model, has_been_replaced = replace_linear_layer_with_target_layer(
+            model,
+            target_cls=self.target_cls,
+            quantization_config=self.quantization_config,
+            modules_to_not_convert=self.modules_to_not_convert,
+        )
+        if not has_been_replaced:
+            logger.warning(
+                "You are loading a GPTQ model but no linear modules were found in your model."
+                " Please double check your model architecture, or submit an issue on GitHub if you think this is a bug."
+            )
+        return model
+
+    def _process_model_after_weight_loading(self, model, **kwargs):
+        """
+        Processes the model after loading weights, repacking quantization layers.
+
+        Args:
+            model (torch.nn.Module): The model to process.
+            **kwargs: Additional keyword arguments.
+
+        Returns:
+            :torch.nn.Module: The processed model.
+        """
+        qlayers = find_layers(model, [QuantLinearGPTQ])
+        for _, qlayer in tqdm.tqdm(qlayers.items(), desc="Repacking qzeros..."):
+            qlayer.handle()
+        print("Done")
```

Comment:
remove `tqdm` and `print`.
we can do this like:
```
for name, module in model.named_modules():
     if isinstance(module, QuantLinearGPTQ):
         module.repack_zeros() 
```
We can remove `find_layers`, not required.